src/xinterpreter.cpp

```diff
@@ -357,7 +359,9 @@ __get_cxx_version ()
 
     void interpreter::init_includes()
     {
+#ifndef EMSCRIPTEN
         Cpp::AddIncludePath((xeus::prefix_path() + "/include/").c_str());
```

Comment:
Does that hurt on emscripten? I'd think it's just some random non-existing folder?

Reply:
Yeah random non existing folder that won't contribute to our solution :)

If you look at Anubhab's solution (https://wasmdemo.argentite.me/) or any other solutions that interest us (https://binji.github.io/wasm-clang/) we are only interested in two folders is what I can see 
1) include ( for `/include/c++/v1`)
2) include/wasm32-emscripten  (for `/include/wasm32-emscripten/c++/v1`)

```
// 1st link
"" -cc1 -triple wasm32-unknown-emscripten -emit-obj -mrelax-all -disable-free -clear-ast-before-backend -main-file-name "<<< inputs >>>" -mrelocation-model static -mframe-pointer=none -ffp-contract=on -fno-rounding-math -mconstructor-aliases -target-cpu generic -fvisibility=hidden -debugger-tuning=gdb -v -fcoverage-compilation-dir=/ -resource-dir lib/clang/17 -internal-isystem /include/wasm32-emscripten/c++/v1 -internal-isystem /include/c++/v1 -internal-isystem lib/clang/17/include -internal-isystem /include/wasm32-emscripten -internal-isystem /include -std=c++17 -fdeprecated-macro -fdebug-compilation-dir=/ -ferror-limit 19 -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fincremental-extensions -o "<<< inputs >>>.o" -x c++ "<<< inputs >>>"
clang -cc1 version 17.0.0 based upon LLVM 17.0.0git default target wasm32-unknown-emscripten
ignoring nonexistent directory "/include/wasm32-emscripten/c++/v1"
ignoring nonexistent directory "lib/clang/17/include"
ignoring nonexistent directory "/include/wasm32-emscripten"
#include "..." search starts here:
#include <...> search starts here:
/include/c++/v1
/include

// 2nd link
clang -cc1 -emit-obj -disable-free -isysroot / -internal-isystem /include/c++/v1 -internal-isystem /include -internal-isystem /lib/clang/8.0.1/include -ferror-limit 19 -fmessage-length 80 -fcolor-diagnostics -O2 -o test.o -x c++ test.cc
```

We eventually need to fetch anything we are interested in from one of the two folders and that is what --preload-file would do for us !

Reply:
Anyways the emscripten Ci would fail for now cause I am playing around with how the shared build for clangCppInteroP would work on emscripten-forge. As I mentioned on discord I wrote my thought on what should be done for cppinterop's emscripten build but I haven't heard anything back on it. Obviously we need to decide on a static build vs a shared build.

Reply:
The question is does not having ifdefs around that code cause a failure on emscripten?